src/flow-hash.c

```diff
@@ -266,38 +267,92 @@ uint32_t FlowKeyGetHash(FlowKey *fk)
         fhk.ports[pi] = fk->dp;
         fhk.proto = (uint16_t)fk->proto;
         fhk.recur = (uint16_t)fk->recursion_level;
-        fhk.vlan_id[0] = fk->vlan_id[0];
-        fhk.vlan_id[1] = fk->vlan_id[1];
+        fhk.vlan_id[0] = fk->vlan_id[0] & g_vlan_mask;
+        fhk.vlan_id[1] = fk->vlan_id[1] & g_vlan_mask;
 
         hash = hashword(fhk.u32, 11, flow_config.hash_rand);
     }
 return hash;
 }
 
+static inline bool cmp_addrs(const uint32_t addr1[4], const uint32_t addr2[4])
```

Comment:
Our code style is that function names AreWrittenLikeThis. See https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Coding_Style

Reply:
I can't remove the vlanh pointers without touching pfring and af-packet because they set those pointers to NULL, but I can try to make sure that that's the only change to those files.